bpf_load.c

```diff
@@ -0,0 +1,669 @@
+// SPDX-License-Identifier: GPL-2.0
```

Comment:
Should add sourcing: https://github.com/MarvellEmbeddedProcessors/mainline-public/blob/master/samples/bpf/bpf_load.c

I spent a while reading this before realizing some of the comments were likely not by you